tests/network/sriov.go

```diff
@@ -0,0 +1,688 @@
+
+				Eventually(func() error {
+					return libnet.PingFromVMConsole(vmi1, ipB)
+				}, 15*time.Second, time.Second).Should(Succeed())
+				Eventually(func() error {
+					return libnet.PingFromVMConsole(vmi2, ipA)
+				}, 15*time.Second, time.Second).Should(Succeed())
+			})
+
+			It("[test_id:3957]should connect to another machine with sriov interface over IPv6", func() {
+				vmi1CIDR := "fc00::1/64"
+				vmi2CIDR := "fc00::2/64"
+				vmi1IP, err := libnet.CidrToIP(vmi1CIDR)
+				Expect(err).ToNot(HaveOccurred())
+				vmi2IP, err := libnet.CidrToIP(vmi2CIDR)
+				Expect(err).ToNot(HaveOccurred())
+
+				//create two vms on the same sriov network
+				vmi1, vmi2 := createSriovVMs(sriovnetLinkEnabled, sriovnetLinkEnabled, vmi1CIDR, vmi2CIDR)
+
+				Eventually(func() error {
+					return libnet.PingFromVMConsole(vmi1, vmi2IP)
+				}, 15*time.Second, time.Second).Should(Succeed())
+				Eventually(func() error {
+					return libnet.PingFromVMConsole(vmi2, vmi1IP)
+				}, 15*time.Second, time.Second).Should(Succeed())
+			})
+
+			Context("With VLAN", func() {
+				const (
+					cidrVlaned1     = "192.168.0.1/24"
+					sriovnetVlanned = "sriov-vlan"
+				)
+				var ipVlaned1 string
+
+				BeforeEach(func() {
+					var err error
+					ipVlaned1, err = libnet.CidrToIP(cidrVlaned1)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(createSriovNetworkAttachmentDefinition(sriovnetVlanned, util.NamespaceTestDefault, sriovVlanConfNAD)).To(Succeed())
+				})
+
+				It("should be able to ping between two VMIs with the same VLAN over SRIOV network", func() {
+					_, vlanedVMI2 := createSriovVMs(sriovnetVlanned, sriovnetVlanned, cidrVlaned1, "192.168.0.2/24")
+
+					By("pinging from vlanedVMI2 and the anonymous vmi over vlan")
+					Eventually(func() error {
+						return libnet.PingFromVMConsole(vlanedVMI2, ipVlaned1)
+					}, 15*time.Second, time.Second).ShouldNot(HaveOccurred())
+				})
+
+				It("should NOT be able to ping between Vlaned VMI and a non Vlaned VMI", func() {
+					_, nonVlanedVMI := createSriovVMs(sriovnetVlanned, sriovnetLinkEnabled, cidrVlaned1, "192.168.0.3/24")
+
+					By("pinging between nonVlanedVMIand the anonymous vmi")
+					Eventually(func() error {
+						return libnet.PingFromVMConsole(nonVlanedVMI, ipVlaned1)
+					}, 15*time.Second, time.Second).Should(HaveOccurred())
+				})
+			})
+		})
+	})
+})
+
+func getInterfaceNameByMAC(vmi *v1.VirtualMachineInstance, mac string) (string, error) {
+	for _, iface := range vmi.Status.Interfaces {
+		if iface.MAC == mac {
+			return iface.InterfaceName, nil
+		}
+	}
+
+	return "", fmt.Errorf("could not get sriov interface by MAC: no interface on VMI %s with MAC %s", vmi.Name, mac)
+}
+
+func getInterfaceNetworkNameByMAC(vmi *v1.VirtualMachineInstance, macAddress string) string {
+	for _, iface := range vmi.Status.Interfaces {
+		if iface.MAC == macAddress {
+			return iface.Name
+		}
+	}
+
+	return ""
+}
```

Comment:
Are those functions also worthy to be moved to `libnet`?

Reply:
They are not used anywhere else but this file.
Some future PR can try to detect where the same is used and centralize it by placing under `libnet`.

Reply:
Thank you.